lib/pry/ring.rb

```diff
@@ -57,10 +57,7 @@ def [](index)
         return @buffer[(count + index) % max_size] if index.is_a?(Integer)
         return @buffer[index] if count <= max_size
 
-        # Swap parts of array when the array turns page and starts overwriting
-        # from the beginning, then apply the range.
-        last_part = @buffer.slice([index.end, max_size - 1].min, count % max_size)
-        (last_part + (@buffer - last_part))[index]
+        to_a[index]
```

Comment:
it seems inefficient to do it like this, can we not access `@buffer` directly instead?